.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,84 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "1.9.4.x", 20.0 ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "1.9.4.x", 20.0 ]
+    paths-ignore:
+      - '**/*.md'
+      - '**/*.txt'
+  schedule:
+    - cron: '33 4 * * 4'
```

Comment:
Do we need this? Set path to `js` only?

Reply:
removed the cron trigger.
We dont have any js files outside the js directory? then makes sense to add it, too

Reply:
we've a lot of .js in /skin/

can it also check the inline js in templates?

Reply:
I'd use include `**.*.js` instead of ... to run it only when js files changed.


```
    paths-ignore:
      - '**/*.md'
      - '**/*.txt'
```